test/testautovariables.cpp

```diff
@@ -3893,6 +3893,17 @@ class TestAutoVariables : public TestFixture {
               "    return { m.data() };\n"
               "}\n");
         ASSERT_EQUALS("", errout_str());
+
+        check("struct S { std::string s; };\n" // #13167
+              "std::vector<S> f() {\n"
+              "    std::vector<S> v;\n"
+              "    {\n"
+              "        std::string a{ \"abc\" };\n"
+              "        v.push_back({ a.c_str() });\n"
+              "    }\n"
+              "    return v;\n"
+              "}\n");
+        ASSERT_EQUALS("", errout_str());
```

Comment:
Do we still get a warning if we change this to std::string_view?

Reply:
No, good catch. Of course, it is nowhere documented what `stdStringLike` actually means.